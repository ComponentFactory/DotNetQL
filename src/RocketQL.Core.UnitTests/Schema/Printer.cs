namespace RocketQL.Core.UnitTests.SchemaValidation;

public class Printer : UnitTestBase
{
    [Theory]
    [InlineData("printer-schema.graphql")]
    public void FileSchema(string filename)
    {
        var schemaText = File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "TestFiles", filename));
        var schema = new Schema();
        schema.Add(schemaText);
        schema.Validate();
        string print = schema.Print();
        print.MatchSnapshot();
    }
}

