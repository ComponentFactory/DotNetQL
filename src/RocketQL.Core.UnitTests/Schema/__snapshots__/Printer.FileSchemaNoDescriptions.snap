schema @directiveAll
{
    query: typeAll
}

type typeAll implements interface1 & interface2 @directiveAll
{
    one: Int
    two: Int @directiveAll
    three
    (
        arg1: Int = 5 @directiveAll
    ): Int
    four
    (
        arg1: compound = { name: "bob", age: 42, alive: true } @directiveAll
    ): Int
}

interface interface1
{
    one: Int
}

interface interface2
{
    two: Int
}

interface interfaceAll implements interface1 & interface2 @directiveAll
{
    one: Int
    two: Int @directiveAll
    three
    (
        arg1: Int = 5 @directiveAll
    ): Int
    four
    (
        arg1: compound = { name: "bob", age: 42, alive: true } @directiveAll
    ): Int
}

union unionAll @directiveAll = typeAll

input compound
{
    name: String
    age: Int
    alive: Boolean
}

input inputAll @directiveAll
{
    one: Int
    two: Int = 5 @directiveAll
    four: compound = { name: "bob", age: 42, alive: true } @directiveAll
}

enum enumAll @directiveAll
{
    FIRST @directiveAll
    SECOND
    THIRD
}

scalar scalarAll @directiveAll

directive @directiveAll repeatable on SCHEMA | SCALAR | OBJECT | FIELD_DEFINITION | ARGUMENT_DEFINITION | INTERFACE | UNION | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION | TYPE_SYSTEM_DRECTIVE_LOCATIONS

